src/tools/web/elektrad/meta_handler.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	elektra "github.com/ElektraInitiative/go-elektra/kdb"
+)
+
+type KeyValueBody struct {
+	Key   string  `json:"key"`
+	Value *string `json:"value"`
+}
+
+func PostMetaHandler(w http.ResponseWriter, r *http.Request) {
+	var meta KeyValueBody
+
+	keyName := parseKeyNameFromURL(r)
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&meta); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	if meta.Key == "" {
+		badRequest(w)
+	}
+
+	kdb := kdbHandle()
+
+	ks, err := elektra.CreateKeySet()
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	parentKey, _ := elektra.CreateKey(keyName)
+	_, err = kdb.Get(ks, parentKey)
+
+	k, err := ks.Lookup(parentKey)
+
+	if k == nil {
+		err = ks.AppendKey(parentKey)
+	}
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	if meta.Value == nil {
+		err = k.DeleteMeta(meta.Key)
+	} else {
+		err = k.SetMeta(meta.Key, *meta.Value)
+
+	}
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	noContent(w)
+}
+
+func DeleteMetaHandler(w http.ResponseWriter, r *http.Request) {
+	// kdb := kdbHandle()
```

Comment:
Functionality missing?

Please try to avoid to always create new handles on every request. Handles are meant to be reused (within a thread of execution).

Reply:
> Please try to avoid to always create new handles on every request. Handles are meant to be reused (within a thread of execution).

Every request is scheduled within a new go routine which can potentially all run on different threads - thats why new handles are created on every request.

Only thing that I can think of is creating a handle pool. But is that really necessary?

Reply:
Yes, this is an important topic which did not work with the previous elektrad.

It is essential that every connected client gets its own KDB handle throughout the whole time. This is not only for performance (it is expensive to create new handles and to call kdbGet on fresh handles) but also for conflict handling: kdbSet can only detect conflicts if the same handle is used throughout the whole session.

qt-gui does this properly, it even is able to do a 3-way merge in case of conflicts. The API for 3-way merging can be found in src/include/kdbmerge.h, so you can already create a binding and use it within elektrad. (Although it is not fully implemented but this does not matter for now.)

You can test easily if it works: simply do modifications with `kdb` while elektrad is running. If elektrad silently overwrites the modifications, then the KDB handles were not used correctly.

Reply:
So my approach would be 
* to set a GUID as a cookie on the first request
* save handles per GUID with a time to live
* retrieve and lock the handle per http request (if the same client sends to request at the same time)

what do you think?

Reply:
Yes, something like this would be needed. Maybe you use something which is already available in Go, something that includes authentication and is more secure than a do-it-yourself solution? What can you propose?

For the website we use https://jwt.io/

Important is, that we still retain the possibility to use elektrad with `curl` or similar tools.

Reply:
So authentication too? I would really like to finish this PR and prefer if we could talk and agree about additional features in another PR.

Reply:
I only asked that you look into existing solutions.

I want to avoid that we hack something which makes problems later.

We can have a phone call to discuss it.

Reply:
Sure - analog phone or some online tool?

Reply:
We can try matrix.org tomorrow after 10:00.

Reply:
Perfekt - koenntest du eventuell bis dahin noch einen blick auf https://github.com/ElektraInitiative/libelektra/pull/2943 werfen? Mag das heute wenn moeglich abschliessen :)

Reply:
Is the GUID cryptographically secured (encryption/signature)?